src/image_int.hpp

```diff
@@ -12,8 +12,10 @@
 #include <ostream>  // for ostream, basic_ostream::put
 #include <string>
 
-#if (defined(__GNUG__) || defined(__GNUC__)) || defined(__clang__)
+#if defined(__clang__)
 #define ATTRIBUTE_FORMAT_PRINTF __attribute__((format(printf, 1, 0)))
+#elif (defined(__GNUG__) || defined(__GNUC__))
+#define ATTRIBUTE_FORMAT_PRINTF __attribute__((format(gnu_printf, 1, 0)))
```

Comment:
How about this instead, more closely reflects the intent:
```
#if defined(__MINGW32__)
#define ATTRIBUTE_FORMAT_PRINTF __attribute__((format(__MINGW_PRINTF_FORMAT, 1, 0)))
#elif defined(__GNUC__) || defined(__clang__)
#define ATTRIBUTE_FORMAT_PRINTF __attribute__((format(printf, 1, 0)))
#else
#define ATTRIBUTE_FORMAT_PRINTF
#endif
```

Reply:
Done.